ACE/ace/Malloc_T.cpp

```diff
@@ -177,6 +177,170 @@ ACE_Dynamic_Cached_Allocator<ACE_LOCK>::free (void * ptr)
     this->free_list_.add ((ACE_Cached_Mem_Pool_Node<char> *) ptr);
 }
 
+template <class ACE_LOCK>
+ACE_Cascaded_Dynamic_Cached_Allocator<ACE_LOCK>::ACE_Cascaded_Dynamic_Cached_Allocator
+  (size_t initial_n_chunks, size_t chunk_size)
+    : initial_n_chunks_ (initial_n_chunks),
+      chunk_size_ (chunk_size)
+{
+  ACE_ASSERT (chunk_size > 0);
+
+  comb_alloc_ptr tmp;
+  ACE_NEW (tmp, comb_alloc_type(this->initial_n_chunks_, this->chunk_size_));
```

Comment:
what when new fails?

Reply:
The instance will lefted in incomplted state, but the incomplted state will be reconstructed in `malloc` API call

and In the same time I add protecting judgement for free API

OK ?